gdal/frmts/hdf5/hdf5multidim.cpp

```diff
@@ -1240,7 +1240,7 @@ bool HDF5Array::ReadSlow(const GUInt64* arrayStartIdx,
             }
             else
             {
-                anStart[i] = arrayStartIdx[i] + arrayStep[i] * (count[i]-1);
+                anStart[i] = arrayStartIdx[i] - (-arrayStep[i]) * (count[i]-1);
```

Comment:
It's confusing what you're trying to achieve here without the context of the PR.  A comment would help.  Also think if parens went around the product it would be easier to read.

Reply:
Parenthesis added. Extra comment added
// Use double negation so that operations occur only on
// positive quantities to avoid an artificial negative signed
// integer to unsigned conversion.